providers/openstack/scs/cluster-class/templates/cluster-class.yaml

```diff
@@ -588,6 +604,34 @@ cre ate group names like oidc:engineering and oidc:infra."
             path: "/spec/template/spec/networkMTU"
             valueFrom:
               variable: network_mtu
+    - name: controlPlaneAvailabilityZones
+      description: "Sets the availability zones which control plane machines may be deployed to."
+      enabledIf: {{ `"{{ if .controlPlaneAvailabilityZones }}true{{end}}"` }}
```

Comment:
I hate helm style: double template-ing, double if-ing, and then you get a patch that you apply in your head in order to get what is going to happen. This is such a broken concept.

Reply:
Yeah, it looks strange, but we have to do it like this for now. The first templating( ``{{`...`}}`` )is for the helm to ignore the second one. Helm is a tool which we use here for the cluster-stacks packaging. And the second one comes directly from CAPI ClusterClass, see https://cluster-api.sigs.k8s.io/tasks/experimental-features/cluster-class/write-clusterclass#optional-patches.

Reply:
I can interpolate this and get what goes where, but it sucks too much energy and is very error-prone without real functional (or at least unit) test